src/Giraffe/Middleware.fs

```diff
@@ -125,4 +130,14 @@ type IServiceCollection with
         this.TryAddSingleton<IJsonSerializer>(NewtonsoftJsonSerializer(NewtonsoftJsonSerializer.DefaultSettings))
         this.TryAddSingleton<IXmlSerializer>(DefaultXmlSerializer(DefaultXmlSerializer.DefaultSettings))
         this.TryAddSingleton<INegotiationConfig, DefaultNegotiationConfig>()
+        this.TryAddTransient<IGiraffeOptions, GiraffeOptions>()
+        this
+
+    member this.AddGiraffe (configureOptions) =
```

Comment:
How about optional:
```suggestion
    member this.AddGiraffe (?configureOptions) =
```

Reply:
This really stings if you use it from C# aspnet core host projects, if we're going with Func and Action etc this should be Optional; DefaultValue(null)

Reply:
Or just add an overload

Reply:
Just so I understand, is the fact it's not optional the bit you object to, or is the optional fsharp parameter suggested above still problematic?

Reply:
Currently the `member this.AddGiraffe (configureOptions)` is an overload. The original `member this.AddGiraffe ()` is still there. Did you guys want me to reduce it to one method with an optional parameter?

Reply:
Nope, that was my bad, what you have here is better 🙂

Reply:
Ok thank you :)

Reply:
Yep overload is much better. If you would need an optional value that borders interop with C# libs it's usually nicer to do `[<Optional;DefaultParameterValue(null)>]` instead of `?param` that way it'll show up as a standard optional value we all know from C#.

Reply:
Oh nice, didn't know that!